File: backend/model/task.py
from pydantic import BaseModel
from typing import Optional

class Task(BaseModel):
    id: int
    user_id: int
    header: str
    detail: str
    is_important: bool
    exp_date: str
    exp_time: str

class Create_Task(BaseModel):
    user_id: int
    header: str
    detail: str
    is_important: bool
    exp_date: str
    exp_time: str


class Task_Create_by_User(BaseModel):
    header: Optional[str] = None
    detail: Optional[str] = None
    is_important: Optional[bool] = None
    exp_date: Optional[str] = None
    exp_time: Optional[str] = None


class Task_Response(BaseModel):
    header: str
    detail: str
    is_important: bool
    

File: frontend/api/task.js
import { apiFetch } from "./auth.js";

export async function get_all() {
    const res = await apiFetch("/task/");
    return res.json()
}

export async function get_one(id) {
    const res = await apiFetch(`/task/${id}`)
    return res.json()
}

export async function create_task(task) {
    const res = await apiFetch("/task",{
        method :"POST",
        body: JSON.stringify({
            id: task.id,
            header: task.header,
            detail: task.description,
            is_important: false,
            exp_date: task.date,
            exp_time: task.time
        })
    })
    return res.json()
}

export async function delete_task(id) {
    const res = await apiFetch(`/task/${id}`,{method: "DELETE"})
    return res.json()
}

export async function modify_task(id, task){
    const res = await apiFetch(`/task/${id}`,{
        method: "PATCH",
        body: JSON.stringify({
            header: task.header,
            detail: task.description,
            is_important: false,
            exp_date: task.date,
            exp_time: task.time
        })
    })
    return res.json()
}

